examples/TTI_codegen.py

```diff
@@ -62,70 +50,26 @@ def damp_boundary(damp):
-        self.m.data[:] = self.model.vp**(-2)
-        self.a = DenseData(name="a", shape=self.model.vp.shape, dtype=self.dtype)
-        self.a.data[:] = 1.0 + 2.0 * self.model.epsilon
-        self.b = DenseData(name="b", shape=self.model.vp.shape, dtype=self.dtype)
-        self.b.data[:] = np.sqrt(1.0 + 2.0 * self.model.delta)
-        self.th = DenseData(name="th", shape=self.model.vp.shape, dtype=self.dtype)
-        self.th.data[:] = self.model.theta
-        if len(dimensions) == 2:
-            self.ph = None
-        else:
-            self.model.phi = np.pad(self.model.phi, tuple(pad_list), 'edge')
-            self.ph = DenseData(name="ph", shape=self.model.vp.shape, dtype=self.dtype)
-            self.ph.data[:] = self.model.phi
-        self.damp = DenseData(name="damp", shape=self.model.vp.shape, dtype=self.dtype)
+        self.damp = DenseData(name="damp", shape=self.model.get_shape_comp(),
+                              dtype=self.dtype)
         # Initialize damp by calling the function that can precompute damping
         damp_boundary(self.damp.data)
-        self.src = SourceLikeTTI(name="src", npoint=1, nt=self.nt, dt=self.dt, h=self.h,
-                                 coordinates=np.array(self.data.source_coords, dtype=self.dtype)[np.newaxis, :],
-                                 ndim=len(dimensions), dtype=self.dtype, nbpml=nbpml)
-        self.rec = SourceLikeTTI(name="rec", npoint=self.nrec, nt=self.nt, dt=self.dt, h=self.h,
-                                 coordinates=self.data.receiver_coords, ndim=len(dimensions), dtype=self.dtype,
-                                 nbpml=nbpml)
-        self.src.data[:] = self.data.get_source()[:, np.newaxis]
-        self.u = TimeData(name="u", shape=self.m.shape, time_dim=self.src.nt, time_order=t_order,
-                          save=False, dtype=self.m.dtype)
-        self.v = TimeData(name="v", shape=self.m.shape, time_dim=self.src.nt, time_order=t_order,
-                          save=False, dtype=self.m.dtype)
-        self.srca = SourceLikeTTI(name="srca", npoint=1, nt=self.nt, dt=self.dt, h=self.h,
-                                  coordinates=np.array(self.data.source_coords, dtype=self.dtype)[np.newaxis, :],
-                                  ndim=len(dimensions), dtype=self.dtype, nbpml=nbpml)
-        if dm_initializer is not None:
-            self.dm = DenseData(name="dm", shape=self.model.vp.shape, dtype=self.dtype)
-            self.dm.data[:] = np.pad(dm_initializer, tuple(pad_list), 'edge')
+        srccoord = np.array(self.data.source_coords, dtype=self.dtype)[np.newaxis, :]
+        self.src = SourceLike(name="src", npoint=1, nt=data.traces.shape[1],
+                              dt=self.dt, h=self.model.get_spacing(),
+                              coordinates=srccoord, ndim=len(self.damp.shape),
+                              dtype=self.dtype, nbpml=nbpml)
+        self.src.data[:] = data.get_source()[:, np.newaxis]
```

Comment:
If the source is passed as aparameter it is added to the data (line 30) so this will also do what it is supposed to